### PR TITLE
ED-1770 create dir profile after creating sso account

### PR DESCRIPTION
**Pull request by kowalcj0**
Best reviewed when #51 is merged
----------------------------------------------

This implements 1st out of 2 scenarios defined in https://uktrade.atlassian.net/browse/ED-1770

```gherkin
    @ED-1770
    @sso
    @fab
    @account
    Scenario Outline: Suppliers with a standalone SSO/great.gov.uk account should be able to select their company for Directory Profile creation
      Given "Peter Alder" has a verified standalone SSO/great.gov.uk account
      And "Peter Alder" is signed in to SSO/great.gov.uk account

      When "Peter Alder" decides to create a trade profile
      And "Peter Alder" randomly selects an active company without a Directory Profile identified by an alias "Company X"
      And "Peter Alder" confirms that "Company X" is the correct one
      And "Peter Alder" confirms that the export status of "Company X" is "<current>"

      Then "Peter Alder" should be prompted to build and improve your Directory Profile

      Examples:
        | current                        |
        | Yes, in the last year          |
        | Yes, 1 to 2 years ago          |
        | Yes, but more than 2 years ago |
        | No, but we are preparing to    |
```

### CODE REVIEW

File: tests/functional/features/steps/fab_when_impl.py [hunk 1]
```diff
@@ -295,33 +284,56 @@ def reg_confirm_export_status(context, supplier_alias, alias, export_status):
...
-                            headers=headers, context=context)
-    expected = ["Create a great.gov.uk account and you can"]
-    check_response(response, 200, strings=expected)
-    logging.debug("Successfully landed on SSO signup page")
-
-    token = extract_csrf_middleware_token(response)
-    context.set_actor_csrfmiddlewaretoken(supplier_alias, token)
     context.export_status = export_status
 
+    if has_sso_account:
```
Comment (konradko) [inline tests/functional/features/steps/fab_when_impl.py]: this is an extremely long function :( Please split it up (e.g. this block could go to a separate one).

File: tests/functional/features/steps/fab_when_impl.py [hunk 2]
```diff
@@ -1066,3 +1079,45 @@ def sso_supplier_confirms_email_address(context, supplier_alias):
...
+    expected = ["Get a trade profile", "Create a trade profile",
+                "Get a trade profile for your company and you can:",
+                "generate new sales leads",
+                "promote your business to thousands of overseas buyers",
+                ("add case studies of your best work to make your company "
+                 "stand out"),
+                ("have buyers contact your sales team directly to get deals "
+                 "moving"),
+                ]
+    check_response(response, 200, strings=expected)
```
Comment (konradko) [inline tests/functional/features/steps/fab_when_impl.py]: `strings` -> `body_contains` could be a little more explicit

File: tests/functional/features/fab/profile.feature [hunk 3]
```diff
@@ -115,3 +115,45 @@ Feature: Trade Profile
...
+    @ED-1770
+    @sso
+    @fab
+    @account
+    Scenario Outline: Suppliers with a standalone SSO/great.gov.uk account should be able to select their company for Directory Profile creation
+      Given "Peter Alder" has a verified standalone SSO/great.gov.uk account
+      And "Peter Alder" is signed in to SSO/great.gov.uk account
+
+      When "Peter Alder" decides to create a trade profile
+      And "Peter Alder" randomly selects an active company without a Directory Profile identified by an alias "Company X"
```
Comment (konradko) [inline tests/functional/features/fab/profile.feature]: is the company getting deleted after running tests? We would avoid the problem of searching for an unregistered active company taking longer over time

Comment (kowalcj0) [inline tests/functional/features/fab/profile.feature]: yes, all the details of the company and the user are deleted from Directory & SSO DBs.
@konradko What would be the fastest way (except using hardcoded company names/numbers) to discover an active company without a Directory profile?

Comment (kowalcj0) [inline tests/functional/features/fab/profile.feature]: DB cleanup queries are here: https://github.com/uktrade/directory-tests/blob/master/tests/functional/features/db_cleanup.py

**Pull request merged**